apps/prairielearn/src/migrations/20231129194831_group_users__group_config_id__add.sql

```diff
@@ -0,0 +1,2 @@
+ALTER TABLE group_users
+ADD COLUMN IF NOT EXISTS group_config_id BIGINT REFERENCES group_configs (id) ON UPDATE CASCADE ON DELETE CASCADE;
```

Comment:
I went back and forth between using `assessment_id` and `group_config_id` here. I ultimately decided on `group_config_id` since that's the nearest "parent" relation, but I'm open to changing this.

Reply:
I'm in favour of using `group_config_id`. I believe the idea of separating the table would allow other non-assessment groups to be formed (e.g., group of students with X extra time in quizzes), and in that case de-coupling groups from assessments when not necessary makes sense. Not sure if this ends up being used in the future, but better to keep it safe.

Reply:
Groups are *already* coupled tightly to assessments - `group_configs` has a mandatory `assessment_id` column and contains assessment-specific configuration. Even the impending `UNIQUE` constraint would make it essentially infeasible to repurpose these tables for anything else.

Reply:
As @jonatanschroeder said, I was also hoping to leave open the possibility of using the groups infrastructure for other purposes in the future. From what I can see we are already able to do this (`group_configs.assessment_id` is nullable)? The changes made in this PR (and the impending UNIQUE) also don't seem to be a problem if we want to use groups in the future that aren't linked to an assessment.

Reply:
Heh, I'm reasonably sure taht `group_configs.assessment_id` is only nullable since it's a) groups code and b) written before we were really cognizant of the benefits of `NOT NULL`). I guess it's a happy accident if that would let us reuse these tables for other kinds of groups (thought I'm not convinced that's a good idea).

An any rate, it doesn't sound like this conversation is relevant to the changes in this PR, so I'll move forward with it!

Reply:
Agreed that this is all irrelevant to this PR!